bika/lims/exportimport/instruments/sysmex/xt/i1800.py

```diff
@@ -0,0 +1,175 @@
+    else:
+        errors.append(t(_("Unrecognized file format ${fileformat}",
+                          mapping={"fileformat": fileformat})))
+
+    if parser:
+        # Load the importer
+        status = ['sample_received', 'attachment_due', 'to_be_verified']
+        if artoapply == 'received':
+            status = ['sample_received']
+        elif artoapply == 'received_tobeverified':
+            status = ['sample_received', 'attachment_due', 'to_be_verified']
+
+        over = [False, False]
+        if override == 'nooverride':
+            over = [False, False]
+        elif override == 'override':
+            over = [True, False]
+        elif override == 'overrideempty':
+            over = [True, True]
+
+        sam = ['getId', 'getSampleID', 'getClientSampleID']
+        if sample == 'requestid':
+            sam = ['getId']
+        if sample == 'sampleid':
+            sam = ['getSampleID']
+        elif sample == 'clientsid':
+            sam = ['getClientSampleID']
+        elif sample == 'sample_clientsid':
+            sam = ['getSampleID', 'getClientSampleID']
+
+        importer = SysmexXTImporter(parser=parser,
+                                  context=context,
+                                  idsearchcriteria=sam,
+                                  allowed_ar_states=status,
+                                  allowed_analysis_states=None,
+                                  override=over,
+                                  instrument_uid=instrument)
+        tbex = ''
+        try:
+            importer.process()
+        except:
+            tbex = traceback.format_exc()
+        errors = importer.errors
+        logs = importer.logs
+        warns = importer.warns
+        if tbex:
+            errors.append(tbex)
+
+    results = {'errors': errors, 'log': logs, 'warns': warns}
+
+    return json.dumps(results)
+
+SEPARATOR = '|'
+SEGMENT_HEADER = 'H'
+SEGMENT_OBSERVATION_ORDER = 'OBR'
+SEGMENT_RESULT = 'OBX'
+SEGMENT_EOF = 'L'
+SEGMENT_COMMENT = 'C'
+
+
+class TX1800iParser(InstrumentTXTResultsFileParser):
+
+    def __init__(self, infile):
+        InstrumentTXTResultsFileParser.__init__(self, infile, SEPARATOR, encoding="utf-8-sig")
+        self._cur_res_id = ''  # Sample ID of the current record
+        self._cur_values = {}  # 'values' dictionary of current sample
+
+    def _parseline(self, line):
+        """
+        All lines come to this method.
+        :param line: a to parse
+        :returns: the number of rows to jump and parse the next data line or
+        return the code error -1
+        """
+        sline = line.split(SEPARATOR)
+        if sline[0] == SEGMENT_HEADER:
+            return self._handle_header(sline)
+        elif sline[0] == SEGMENT_OBSERVATION_ORDER:
+            return self._handle_new_record(sline)
+        elif sline[0] == SEGMENT_RESULT:
+            return self._handle_result_line(sline)
+        elif sline[0] == SEGMENT_EOF:
+            return self._handle_eof(sline)
```

Comment:
You are accessing here multiple times the first element of the header, therefore I would create a variable like `segment = sline[0]` and then you could think of a mapping of segment -> handler to get it immediately, e.g.
```python
handlers = {
    SEGMENT_HEADER: self._handle_header,
    ...
}
handler = hanlers.get(segment)
...
```